tools/src/main/scala/caliban/tools/stitching/RemoteQuery.scala

```diff
@@ -0,0 +1,65 @@
+package caliban.tools.stitching
+
+import caliban.{ GraphQLRequest, InputValue }
+import caliban.execution.Field
+import caliban.Value._
+import caliban.InputValue._
+import caliban.Value.FloatValue._
+import caliban.Value.IntValue._
+
+case class RemoteQuery(field: Field) { self =>
+  def toGraphQLRequest: GraphQLRequest =
+    GraphQLRequest(query =
+      Some(
+        RemoteQuery.QueryRenderer.render(self)
+      )
+    )
+}
+
+object RemoteQuery {
+  object QueryRenderer {
+    def render(r: RemoteQuery): String = s"query { ${renderField(r.field)} }"
+
+    private def renderField(field: Field): String = {
+      val children = renderFields(field)
+      val args     = renderArguments(field.arguments)
+      val alias    = field.alias.map(a => s"$a: ").getOrElse("")
+      val str      = s"$alias${field.name}$args$children"
+
+      withCondition(field)(str)
+    }
+
+    private def withCondition(f: Field)(inner: String): String =
+      f.condition.flatMap(_.headOption).map(x => s"...on $x { $inner }").getOrElse(inner)
+
+    private def renderInputValue(v: InputValue): String =
+      v match {
+        case StringValue(value)      => s""""$value""""
```

Comment:
Can we use the `toString` implementation of `Value` or is there any difference?

Reply:
There were some differences for some other other ones if I'm not mistaken. And it felt a bit weird to me to  only use `toString` for some of the fields. But can revisit if you want to!

Reply:
Ah that's fine, I only checked quickly and couldn't spot any difference 😅